tests/autoblocks/test_autoblocks_configs.py

```diff
@@ -0,0 +1,249 @@
+                my_val="val-from-remote",
+            ),
+        ),
+    )
+
+    config = MyConfig(
+        value=MyConfigValue(my_val="initial-val"),
+    )
+    config.activate_from_remote(
+        config=LatestRemoteConfig(id="my-config-id", latest=True), parser=MyConfigValue.model_validate
+    )
+
+    assert config.value == MyConfigValue(my_val="val-from-remote")
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "AUTOBLOCKS_API_KEY": "mock-api-key",
+    },
+)
+def test_activates_specific_version(httpx_mock):
+    httpx_mock.add_response(
+        url=f"{API_ENDPOINT}/configs/my-config-id/versions/1",
+        method="GET",
+        match_headers={"Authorization": "Bearer mock-api-key"},
+        json=dict(
+            id="my-config-id",
+            version="1",
+            value=dict(
+                my_val="val-from-remote",
+            ),
+        ),
+    )
+
+    config = MyConfig(
+        value=MyConfigValue(my_val="initial-val"),
+    )
+    config.activate_from_remote(
+        config=RemoteConfigWithVersion(id="my-config-id", version="1"), parser=MyConfigValue.model_validate
+    )
+
+    assert config.value == MyConfigValue(my_val="val-from-remote")
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "AUTOBLOCKS_API_KEY": "mock-api-key",
+    },
+)
+def test_activates_undeployed_latest(httpx_mock):
+    httpx_mock.add_response(
+        url=f"{API_ENDPOINT}/configs/my-config-id/revisions/latest",
+        method="GET",
+        match_headers={"Authorization": "Bearer mock-api-key"},
+        json=dict(
+            id="my-config-id",
+            version="1",
+            value=dict(
+                my_val="val-from-remote",
+            ),
+        ),
+    )
+
+    config = MyConfig(
+        value=MyConfigValue(my_val="initial-val"),
+    )
+    config.activate_from_remote(
+        config=DangerouslyUseUndeployedRemoteConfig(
+            id="my-config-id", dangerously_use_undeployed=LatestDangerouslyUseUndeployed(latest=True)
+        ),
+        parser=MyConfigValue.model_validate,
+    )
+
+    assert config.value == MyConfigValue(my_val="val-from-remote")
+
+
+@mock.patch.dict(
+    os.environ,
+    {
+        "AUTOBLOCKS_API_KEY": "mock-api-key",
+    },
+)
+def test_activates_undeployed_revision_id(httpx_mock):
+    httpx_mock.add_response(
+        url=f"{API_ENDPOINT}/configs/my-config-id/revisions/my-revision-id",
+        method="GET",
+        match_headers={"Authorization": "Bearer mock-api-key"},
+        json=dict(
+            id="my-config-id",
+            version="1",
+            value=dict(
+                my_val="val-from-remote",
+            ),
+        ),
+    )
+
+    config = MyConfig(
+        value=MyConfigValue(my_val="initial-val"),
+    )
+    config.activate_from_remote(
+        config=DangerouslyUseUndeployedRemoteConfig(
+            id="my-config-id",
+            dangerously_use_undeployed=DangerouslyUseUndeployedWithRevision(revision_id="my-revision-id"),
+        ),
```

Comment:
seeing this written out it's pretty verbose, and will also require you to make quite a few temp code changes locally to use a local revision

Reply:
ya i was thinking that too for python we could do it all top level so like

DangerouslyUseUndeployedRemoteConfigWithRevision ?

Reply:
update this, I think it does feel nicer, lmk what you think